packages/hardhat-chai-matchers/src/withArgs.ts

```diff
@@ -1,9 +1,39 @@
+import { AssertionError } from "chai";
+
+import { isBigNumber, normalizeToBigInt } from "hardhat/common";
+
 import { emitWithArgs, EMIT_CALLED } from "./emit";
 import {
   revertedWithCustomErrorWithArgs,
   REVERTED_WITH_CUSTOM_ERROR_CALLED,
 } from "./reverted/revertedWithCustomError";
 
+export function anyValue(): boolean {
+  return true;
+}
+
+export function anyUint(i: any): boolean {
+  if (typeof i === "number") {
+    if (i < 0) {
+      throw new AssertionError(
+        `anyUint expected its argument to be an unsigned integer, but it was negative, with value ${i}`
+      );
+    }
+    return true;
+  } else if (isBigNumber(i)) {
```

Comment:
This code would be simpler if `isBigNumber` returned true for numbers (although the name wouldn't make sense), because `normalizeToBigInt` does accept numbers.

Nothing actionable here, just something to keep in mind, especially if we make further modifications to our number helpers.